### PR TITLE
Automate docs with eslint-doc-generator

### PR DESCRIPTION
**Pull request by bmish**
I built this CLI tool [eslint-doc-generator](https://github.com/bmish/eslint-doc-generator) for automating the generation of the README rules list and rule doc title/notices for ESLint plugins. It follows common documentation conventions from this and other top ESLint plugins and will help us standardize documentation across ESLint plugins (and generally improve the usability of our custom rules through better documentation and streamline the process of adding new rules).

eslint-doc-generator is significantly more robust compared to the previous documentation tests which I have now removed. It has additional functionality (for example, the rules list legend is auto-generated, and the rules list will show additional columns of information when applicable) and 100% test coverage.

I recently added eslint-doc-generator to the related plugin eslint-plugin-unicorn in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1930.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,7 +93,7 @@
 			"strict": "error",
 			"ava/no-ignored-test-files": "off",
 			"eslint-plugin/prefer-message-ids": "off",
-			"eslint-plugin/require-meta-docs-description": "off",
+			"eslint-plugin/require-meta-docs-description": ["error", { "pattern": "^(Enforce|Ensure|Require|Disallow|Prevent|Prefer)" }],
```
Comment (bmish) [inline package.json:96]: We now enforce that rule descriptions are stored inside the standard `meta.docs.description` field in rules. This allows tooling like eslint-doc-generator to utilize these descriptions. I made the same change in eslint-plugin-unicorn last year in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1155.

File: readme.md [hunk 2]
```diff
@@ -34,71 +34,53 @@ Configure it in `package.json`.
...
-			"ava/test-title": "error",
-			"ava/test-title-format": "off",
-			"ava/use-t": "error",
-			"ava/use-t-throws-async-well": "error",
-			"ava/use-t-well": "error",
-			"ava/use-test": "error",
-			"ava/use-true-false": "error"
-		}
-	}
+			"ava/...": "error"
```
Comment (bmish) [inline readme.md:37]: This matches https://github.com/sindresorhus/eslint-plugin-unicorn#usage

**Review by sindresorhus — APPROVED** [review]

**Comment by sindresorhus** [comment]
@novemberborn Not sure what assigning it to me means?

**Comment by novemberborn** [comment]
@sindresorhus please review, merge, ship… whatever you like. I don't have enough context here.

**Comment by bmish** [comment]
By the way, since Node 12 is no longer supported (https://github.com/avajs/eslint-plugin-ava/commit/bd8c4c6e6c20ccce948df6f72e8d02c4e7f4080f), an admin will need to remove the Node 12 check from this repo's branch protection settings.

**Pull request merged**